update_user_agents.py

```diff
@@ -0,0 +1,352 @@
+        """Get the browser as a string."""
+        browser = user_agent.browser.family
+
+        if re.search(self.chrome_pattern, browser):
+            return "chrome"
+        elif re.search(self.firefox_pattern, browser):
+            return "firefox"
+        elif re.search(self.safari_pattern, browser):
+            return "safari"
+        elif re.search(self.edge_pattern, browser):
+            return "edge"
+
+        return browser
+
+    def _get_os(self, user_agent: UserAgent) -> str:
+        """Get the operating system as a string."""
+        os = user_agent.os.family
+        if re.search(self.ios_pattern, os):
+            return "ios"
+        elif re.search(self.linux_pattern, os):
+            return "linux"
+        elif re.search(self.macos_pattern, os):
+            return "macos"
+        elif re.search(self.windows_pattern, os):
+            return "win10"
+        elif re.search(self.android_pattern, os):
+            return "android"
+
+        return os
+
+    @staticmethod
+    def _get_version(user_agent: UserAgent) -> float:
+        "Get the browser version as a float."
+        version_groups = user_agent.browser.version_string.split(".")
+        if len(version_groups) == 0 or version_groups[0] == "":
+            return 0.0
+        if len(version_groups) == 1:
+            return float(version_groups[0])
+        else:
+            return float(".".join(version_groups[:2]))
+
+
+if __name__ == "__main__":
+    updater = UserAgentUpdater(".")
+
+    # Parse the command line arguments, if provided.
+    parser = argparse.ArgumentParser(
+        description="Script to update the list of user agents."
+    )
+    parser.add_argument(
+        "browser_args",
+        nargs="*",
+        default=[],
+        help="Arbitrary number of browsers to include in the update",
+    )
+    parser.add_argument(
+        "--limit",
+        default=200,
+        type=int,
+        help="The max number of user agents per browser",
+    )
+    parser.add_argument(
+        "--max_version_lag",
+        type=float,
+        nargs="*",
+        default=0.0,
+    )
+
+    args = parser.parse_args()
+    possible_browsers = [
+        "chrome",
+        "edge",
+        "safari",
+        "firefox",
+        "samsung-browser",
+    ]
+
+    requested_browsers = [
+        browser for browser in args.browser_args if browser in possible_browsers
+    ]
+    max_version_lags = args.max_version_lag
+    if not isinstance(max_version_lags, list):
+        max_version_lags = [max_version_lags]
+    limit = args.limit
+
+    if len(requested_browsers) > 0:
+        for requested_browser, max_version_lag in zip(
+            requested_browsers, cycle(max_version_lags)
+        ):
+            updater.get_user_agents(
+                browser_type=requested_browser,
+                limit=limit,
+                remember=True,
+                max_version_lag=max_version_lag,
+            )
```

Comment:
Adding a time sleep to avoid getting rate limited or blocked by the server. Be nice for the server.